pkg/logger/logger.go

```diff
@@ -0,0 +1,119 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type loggerState byte
+
+const (
+	new loggerState = iota
+	started
+	ended
+)
+
+type Logger struct {
+	filename string
+	state    loggerState
+}
+
+func New(filename string) Logger {
+	return Logger{filename, new}
+}
+
+func (logger *Logger) logEvent(event map[string]interface{}) error {
+	jsonData, err := json.Marshal(event)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(logger.filename, os.O_APPEND|os.O_WRONLY, 0644)
```

Comment:
I see that you made some changes which does alleviate some concerns (opening file each time a line is logged) but not all - I was hoping that the new design would not require I/O usage in tests since you could use it with an in-memory buffer but the current design still doesn't allow it as the `Logger` does not have a public constructor.

Reply:
There are two loggers now: Logger, which operates on an `io.Writer`, and FileLogger, which operates on files and uses the first Logger.
I added a public constructor, but I'm not sure about one thing: since we can do `logger.Logger{writer}`, is a constructor like `logger.New(writer)` strictly necessary? They both do the same thing after all, and the first one was already public.

Reply:
Added test for regular Logger, which uses `bytes.Buffer` instead of a file